### ISSUE TITLE
Antigen doesn't work with plugins whose repositories don't have a 'master' branch

### ISSUE THREAD

**Issue by FredDeschenes** [zsh-users/antigen#717]
Description
-----------

Title pretty much says it all. Antigen seems to rely on the fact that a repository has a 'master' branch and with Github's new default repository setup that uses 'main' instead (or any repository that changed their 'master' branch to 'main') installing/updating those plugins fails.

Steps to reproduce
------------------

Minimal steps to reproduce this behavior. Example:

    1 - Use any theme/plugin that doesn't have a 'master' branch ([the theme I use](https://github.com/reobin/typewritten) is the one that was giving me issues)
    2 - Run `antigen update`
        $ antigen update
        Updating reobin/typewritten@master... Error! Activate logging and try again.

Expected behavior:

    - Plugins without a 'master' branch should work
 

## Software version

- `antigen version`
Antigen develop (d1dd78b)
Revision date: 2018-01-15 14:37:21 -0300
- `zsh --version`
zsh 5.8 (x86_64-ubuntu-linux-gnu)
- `uname -a`
Linux poste-0948 5.4.0-51-generic #56-Ubuntu SMP Mon Oct 5 14:28:49 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

## Configuration

- gist of `.antigenrc`
Not used
- gist of `.zshrc`
Entire file is [here](https://github.com/FredDeschenes/dotfiles/blob/master/zsh/.zshrc#L69), but line 69 is where my theme that causes the issue is set.

## More information

As a workaround you can manually clone/update your plugins in $ANTIGEN_BUNDLES (Antigen still creates the directory structure, just doesn't perform the clone/pull).

**Comment by hermitmaster** [comment]
For anyone that runs into this problem, you can specify a branch or tag. See the wiki.

**Comment by acidghost** [comment]
@FredDeschenes This is surely late, but you can specify a branch or tag you want to clone with `antigen bundle` (RTFM https://github.com/zsh-users/antigen/wiki/Commands).

Example: `antigen bundle org/repo@main`

**Comment by reegnz** [comment]
Would it make sense for antigen to just not assume the default branch of the repo is called `master`?
It's somewhat user-hostile to have the user figure out what the default branch of a repo is, instead of infering it from the repo itself.
Having any branch as the default branch is a git feature.

**Comment by philoserf** [comment]
FTR: Commonly used plugins from @unixorn have moved to main

**Comment by unixorn** [comment]
FWIW, you can detect the default branch with

```
git remote show origin | awk '$0 ~ \"HEAD branch\" {print $NF}'
```

**Comment by dylan-chong** [comment]
Would be really nice if at least we could have by default `main` as an accepted alternative to `master`

**Comment by aredridel** [comment]
Yes please! Especially since that's the default on modern git and github

**Comment by unixorn** [comment]
I think it's best to use `git remote show origin | awk '$0 ~ \"HEAD branch\" {print $NF}'` to find a given repo's default branch. Otherwise there will still be an issue when you come across a repo that used something like `primary` instead of `main` when they moved off of `master`.

**Comment by gko** [comment]
There is also a problem with already installed bundles. If you change the branch from `master` to `main` you have to update your config.

There is a way to get a default branch in git repo:
https://github.com/gko/dotfiles/blob/491cb022e07e8ba20cd8cb90fce0fdaebc2439b0/.gitconfig#L62
